lib/api/controller/base.js

```diff
@@ -632,13 +632,31 @@ class NativeController extends BaseController {
     return null;
   }
 
+  /**
+  * Returns the search body query according to the http method
+  *
+  * @param {Request} request
+  *
+  * @returns {Object}
+  */
+  getSearchBody (request) {
+    if ( request.context.connection.protocol === 'http'
+      && request.context.connection.misc.verb === 'GET'
+      && request.input.args.searchBody
+    ) {
+      return JSON.parse(request.input.args.searchBody);
```

Comment:
Parsing exceptions should be caught and converted into `api.assert.invalid_argument` errors.